linchpin/cli/__init__.py

```diff
@@ -405,6 +405,29 @@ def lp_up(self, targets=(), run_id=None, tx_id=None, inv_f="cfg"):
 
 
 
+    def lp_validate(self, targets=(), old_schema=False):
+        """
+        This function takes a list of targets, and validates their topology.
+
+        :param targets:
+            A tuple of targets to provision
+ 
+        :param old_schema
+            Denotes whether schema should be validated with the old schema
+            rather than the new one!/usr/bin/env python
+        """
+
+        # Prep input data
+
+        # Execute prepped data
+        return_code, return_data = self._execute_action('validate', targets,
+                                                        old_schema=old_schema)
```

Comment:
I see where you are going here, but I don't consider 'validate' an action. There are only two actions at the moment, 'up' and 'destroy'. Validation would be considered a component of either of these actions. Maybe we should figure out a better way to perform validation and not modify the execute_action function? I'd think we would want the validation to happen *before* executing an up or down. It is its own method also, so calling lp_validate here makes logically more sense.

Reply:
That's fine with me.  Without this pr, validation occurs inside of either _execute(), which is called by _execute_action() so I'll need to decouple a bit more

Reply:
Yeah, adding a call in each of those to your validation seems the right approach. We could definitely decouple more. There's a lot of complexity in those two functions, so it's not super straightforward and probably should be refactored anyway.